### PR TITLE
feat: api handle redirects (#3246)

### PR DESCRIPTION
**Pull request by eddiejaoude**
* feat: api handle redirects

* broken code

* comment out tests

* tests

* tests

* remove

* directory for username

* directory for links

* api redirect

* changes

* Restore package-lock.json

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/3502"></a>

### CODE REVIEW

File: pages/api/users/[username]/links/[url].js [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
+  let data = {};
+  try {
+    data = JSON.parse(fs.readFileSync(filePath, "utf8"));
+  } catch (e) {
+    console.log(`ERROR loading username ${username}`, e);
+    return res.status(404).json({ error: `ERROR ${username} not found` });
+  }
+
+  if (!data.links.find((link) => link.url === url)) {
+    console.log(`ERROR link ${url} not found for username ${username}`);
```
Comment (krshkun) [inline pages/api/users/[username]/links/[url].js:31]: We also use this endpoint for socials 😅

Comment (eddiejaoude) [reply pages/api/users/[username]/links/[url].js:31]: Good spot 👍 

(oops I hit resolved instead of comment, time for me to take a break)

Comment (eddiejaoude) [reply pages/api/users/[username]/links/[url].js:34]: I fixed with a `&&` condition, please can you re-review

**Review by krshkun — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**